datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -579,11 +578,20 @@ impl StreamedBatch {
     }
 }
 
+/// The data stored in this BufferedBatch
```

Comment:
Using an enum it becomes clear in the code that only one of these two states is possible